src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1075,7 +1075,28 @@ protected BoundExpression BindInferredVariableInitializer(DiagnosticBag diagnost
             }
 
             diagnostics.AddRangeAndFree(localDiagnostics);
-            var boundDeclType = new BoundTypeExpression(typeSyntax, aliasOpt, inferredType: isVar, type: declTypeOpt.TypeSymbol);
+
+            ImmutableArray<BoundExpression> dimensionsOpt;
+            if (typeSyntax is ArrayTypeSyntax arrayTypeSyntax && arrayTypeSyntax.RankSpecifiers.Count > 0)
+            {
+                var sizes = ArrayBuilder<BoundExpression>.GetInstance();
+                // We don't want to report any errors from this, as the syntax is invalid anyway
+                var dummyDiagnostics = DiagnosticBag.GetInstance();
+                foreach (var expressionSyntax in arrayTypeSyntax.RankSpecifiers[0].Sizes)
```

Comment:
>RankSpecifiers[0] [](start = 65, length = 17)

What about other rank specifiers?

Reply:
As above

> In an array creation that would also be invalid. For now I'm just trying to capture cases where the syntax would be valid in an ArrayCreationExpression, but invalid in a variable declaration, as I feel these are probably the most common cases.
>
> I think there a huge amount of scope to do more here. This is meant to be an initial pull request with the minimum changes required to be useful.

Reply:
Is it important that we distinguish in the bound tree and the Semantic Model between:

```csharp
int[10,][] x;
int[,10][] x;
int[,][10] x;
```
?

If so we will need to add a new type of BoundNode which contains this information as well as the BoundExpression for the invalid dimensions.

Reply:
>Is it important that we distinguish in the bound tree and the Semantic Model between:
>
>csharp
int[10][] x;
int[][10] x;

I cannot think of any need for this at the moment. The sizes are misplaced expressions.

---
In reply to: [253308442](https://github.com/dotnet/roslyn/pull/32843#discussion_r253308442) [](ancestors = 253308442)